include/xtensor-r/rarray.hpp

```diff
@@ -24,14 +24,15 @@
 
 namespace xt
 {
-
     template <class T>
     class rarray;
 
     template <class T>
     struct xcontainer_inner_types<rarray<T>>
     {
-        using storage_type = xbuffer_adaptor<T*>;
+        using r_type = T;
```

Comment:
Do we need to define this? If not, I would suggest sticking to `T` because calling it `r_type` just leads to confusion as e.g. the `Rcpp` snippet defines `rtype = LGLSXP`, so that `rtype` has a different meaning.

Reply:
I think we do? It's used over in `rcontainer` for the error messaging. I don't know how to pull out that `r_type` directly from `rcontainer` otherwise.

Meaning:

rcontainer sees: `rcontainer<D>` where `D = rarray<T>`. How would I access `T`? By using `r_type` I've stored `T` so that `rcontainer` can see it. 

I'm ignorant though so I'd be happy to do it a different way if you want